packages/flame_spine/lib/src/skeleton_render.dart

```diff
@@ -0,0 +1,174 @@
+import 'dart:convert';
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:spine_core/spine_core.dart';
+import 'package:spine_flutter/spine_flutter.dart';
+
+class SkeletonRender {
+  SkeletonRender({
+    required SkeletonAnimation skeleton,
+    BoxFit boxFit = BoxFit.contain,
+    Alignment alignment = Alignment.center,
+    PlayState playState = PlayState.playing,
+    bool debugRendering = false,
+    bool triangleRendering = true,
+    double frameSizeMultiplier = 0.0,
+    String? animation,
+  }) : _renderObject = SkeletonRenderObject()
+          ..skeleton = skeleton
+          ..fit = boxFit
+          ..alignment = alignment
+          ..playState = playState
+          ..debugRendering = debugRendering
+          ..triangleRendering = triangleRendering
+          ..frameSizeMultiplier = frameSizeMultiplier
+          ..animation = animation;
+
+  late final SkeletonRenderObject _renderObject;
+
+  SkeletonAnimation get skeleton => _renderObject.skeleton;
+
+  set skeleton(SkeletonAnimation value) {
+    _renderObject.skeleton = value;
+  }
+
+  Alignment get alignment => _renderObject.alignment as Alignment;
+
+  set alignment(AlignmentGeometry value) {
+    _renderObject.alignment = value as Alignment;
+  }
+
+  BoxFit get boxFit => _renderObject.fit;
+
+  set boxFit(BoxFit value) {
+    _renderObject.fit = value;
+  }
+
+  PlayState get playState => _renderObject.playState;
+
+  set playState(PlayState value) {
+    _renderObject.playState = value;
+  }
+
+  bool get debugRendering => _renderObject.debugRendering;
+
+  set debugRendering(bool value) {
+    _renderObject.debugRendering = value;
+  }
+
+  bool get triangleRendering => _renderObject.triangleRendering;
+
+  set triangleRendering(bool value) {
+    _renderObject.triangleRendering = value;
+  }
+
+  /// Нow many percent increase the size of the animation
+  /// relative to the size of the first frame.
+  double get frameSizeMultiplier => _renderObject.frameSizeMultiplier;
+
+  set frameSizeMultiplier(double value) {
+    _renderObject.frameSizeMultiplier = value;
+  }
+
+  /// A start animation. We will use it for calculate bounds by frames.
+  String? get animation => _renderObject.animation;
+
+  set animation(String? value) {
+    _renderObject.animation = value;
+  }
+
+  void advance(double dt) {}
```

Comment:
How come this is not doing anything?

Reply:
advancing animation is handled by SkeletonRenderObject internally, so when we set playstate to playing it will call update inside render object.

the above signature is just to preserve standards.

Reply:
What do you mean with "preserve standards", seems unnecessary to have an empty method that can't affect anything? But also, if it is ticking forward itself within the render object the animation won't stop when the engine is paused, which might be a bit problematic. Do they not expose the advance method in the render object?

Reply:
nope render object doesn't expose.
skeleton object has a similar advance called `updateState`.
if call it inside advance animation goes at twice speed since its called twice, once in our render and once in their renderObject.

Reply:
implemented advance method and also updated example to have pause enginge

Reply:
I don't think it is worth implementing advance if you can only control the skeleton, it will be more confusing than it will help anyone, better to just remove it completely.

Reply:
yup, i was also thinking same,
removed that from here. i think we don't need that advance method.
i have updated example to have pause engine and it work fine

Reply:
should i completely remove advance  ?

Reply:
> should i completely remove advance  ?

Yes, if it is empty it should be removed.

Just to clarify, when you had the skeleton in the paused state and drove the updateState yourself, was the animation still moving if you didn't call updateState (and animated properly when it was called)? That could be a good work around if that worked.

Reply:
So if updateState was not called it was frozen?

Reply:
Yes,when we set play state to pause it will stop calling update state and animation will be frozen.